### PR TITLE
Fix/dw bugs

### PR DESCRIPTION
**Pull request by javadkh2**
### Modify this title

Related Issue/Asana ticket: \_

Short description: \_

### Test scenarios

- description of the (manually) executed test scenarios

### Reminders (if applicable)

- [ ] I ran `pnpm install` and `pnpm test` in the root of the monorepo
      (optionally with `--filter=...package...` to exclude non-affected
      projects)
- [ ] I ran `pnpm changeset` in the root of the monorepo
- [ ] Test coverage has not decreased
- [ ] Docs have been updated to reflect changes in PR (don't forget
      docs.kadena.io)

### CODE REVIEW

File: packages/apps/dev-wallet/src/pages/wallet-recovery/Components/RecoveredV3.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ export function RecoveredV3({
   cancel: () => void;
 }) {
   const [error, setError] = useState<string | undefined>();
-  const navigate = useNavigate();
+  const navigate = useTheCorrectNavigate();
```
Comment (sstraatemans) [inline packages/apps/dev-wallet/src/pages/wallet-recovery/Components/RecoveredV3.tsx:40]: naming things 😅

File: packages/apps/dev-wallet/src/App/NavigationContext.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+const context = createContext<INavigationContext>({
+  route: null,
+  navigate: () => {
+    throw new Error('No navigation provider');
+  },
+});
+
+// the useNavigate hook from react-router-dom doesn't work correctly in some cases so I did this workaround to fix it
```
Comment (sstraatemans) [inline packages/apps/dev-wallet/src/App/NavigationContext.tsx:15]: can you add a link or something to the more detail bug?
for future reference

File: packages/apps/dev-wallet/src/App/Layout/Layout.tsx [hunk 3]
```diff
@@ -26,7 +27,7 @@ import {
 export const Layout: FC = () => {
   const { theme, setTheme } = useTheme();
   const location = useLocation();
-  const navigate = useNavigate();
+  const navigate = useTheCorrectNavigate();
```
Comment (alber70g) [inline packages/apps/dev-wallet/src/App/Layout/Layout.tsx:30]: Why is this name like that 😂?

```suggestion
  const navigate = usePatchedNavigate();
```

File: packages/apps/dev-wallet/src/modules/db/db.service.tsx [hunk 4]
```diff
@@ -59,13 +27,12 @@ export const setupDatabase = execInSequence(async (): Promise<IDBDatabase> => {
...
     if (oldVersion === 0) {
       console.log('creating new database');
       createTables(db);
-      return db;
-    }
-    if (oldVersion < 37) {
+    } else if (oldVersion < 37) {
       const confirmed = confirm(
         'You’re using an outdated database version that doesn’t support migration. To continue using the app, all data must be wiped. Do you want to proceed?',
       );
```
Comment (alber70g) [inline packages/apps/dev-wallet/src/modules/db/db.service.tsx:38]: I saw the [conf](https://github.com/kadena-community/kadena.js/pull/2709/files#diff-d4909648622d751e781bd4ca43cabe1e48da9213bffe170509c3c8a3b4a0b3d5) has a `DB_WIPE_ON_VERSION_CHANGE: true`, is this working together?

**Review by alber70g — APPROVED** [review]

File: packages/apps/dev-wallet/src/App/Layout/Layout.tsx (hunk 3 above)
Comment (javadkh2) [reply packages/apps/dev-wallet/src/App/Layout/Layout.tsx:30]: I was about to name it `useTheFuckingNavigation` but changed it at the last moment ;)

File: packages/apps/dev-wallet/src/modules/db/db.service.tsx (hunk 4 above)
Comment (javadkh2) [reply packages/apps/dev-wallet/src/modules/db/db.service.tsx:38]: no the DB_WIPE_ON_VERSION_CHANGE option is obsolete now, I'll remove it

**Pull request merged**